pkg/controller/operator.go

```diff
@@ -105,15 +104,26 @@ func (s *Operator) Run(ctx context.Context, controller *controllercmd.Controller
 	// the last sync time, if any was set
 	statusReporter := status.NewController(configClient, gatherKubeClient.CoreV1(), configObserver, os.Getenv("POD_NAMESPACE"))
 
+	var anonymizer *anonymization.Anonymizer
+	if anonymization.IsObfuscationEnabled(configObserver) {
+		// anonymizer is responsible for anonymizing sensitive data, it can be configured to disable specific anonymization
+		anonymizer, err = anonymization.NewAnonymizerFromConfigClient(ctx, kubeClient, configClient)
+		if err != nil {
+			return err
```

Comment:
I am still not sure if we should fail or just log an error. This still can theoretically lead to crash loop.

Reply:
Well, if obfuscation is disabled `anonymization.IsObfuscationEnabled` will return false and we won't reach here, but we can also just log an error in case it's enabled.

Reply:
Yeah sure, but when it's enabled and some of the required resources (e.g DNS) is not available -> crash & problem :)

Reply:
changed to logging an error